extensions/sql-database-projects/src/test/project.test.ts

```diff
@@ -177,6 +177,29 @@ describe('Project: sqlproj content operations', function (): void {
 		should(project.databaseReferences.length).equal(3);
 		should(project.databaseReferences[2]).equal('test');
 	});
+
+	it('Should not allow adding duplicate database references', async function (): Promise<void> {
+		projFilePath = await testUtils.createTestSqlProjFile(baselines.newProjectFileBaseline);
+		const project = new Project(projFilePath);
+		await project.readProjFile();
+
+		should(project.databaseReferences.length).equal(0);
+		await project.addSystemDatabaseReference(SystemDatabase.master);
+		should(project.databaseReferences.length).equal(1);
+		should(project.databaseReferences[0]).equal(constants.master);
+
+		// try to add reference to master again
+		await testUtils.shouldThrowSpecificError(async () => await project.addSystemDatabaseReference(SystemDatabase.master), constants.databaseReferenceAlreadyExists);
+		should(project.databaseReferences.length).equal(1);
```

Comment:
With this many should checks make sure you add a message for each one so if it fails you know which one did. Right now it would just fail with a generic `expected : 1 actual : #` which isn't terribly useful.